app/uk/gov/hmrc/helptosavefrontend/controllers/RegisterController.scala

```diff
@@ -154,11 +165,12 @@ class RegisterController @Inject() (val helpToSaveService:     HelpToSaveService
             // user has gone through journey already this sessions and were found to be ineligible
             _ ⇒ SeeOther(routes.EligibilityCheckController.getIsNotEligible().url),
             userInfo ⇒
-              // user has gone through journey already this sessions and were found to be eligible
-              userInfo.userInfo.email.fold(ifEligibleWithoutEmail(EligibleWithNoEmail(userInfo.userInfo, userInfo.eligible)))(email ⇒
+              // user has gone through journey already this sessions and were found to be eligible,
+              // but they might not have gone through the email journey, so check if the email is valid or not
+              userInfo.userInfo.email.fold(ifEligibleWithoutEmail(EligibleWithNoEmail(userInfo.userInfo, userInfo.eligible, session.bankDetails)))(email ⇒
                 emailValidation.validate(email).toEither match {
-                  case Right(validEmail) ⇒ ifEligibleWithEmail(EligibleWithEmail(userInfo.userInfo, validEmail, session.confirmedEmail, userInfo.eligible))
-                  case Left(_)           ⇒ ifEligibleWithoutEmail(EligibleWithNoEmail(userInfo.userInfo, userInfo.eligible))
+                  case Right(_) ⇒ ifEligibleWithEmail(EligibleWithEmail(userInfo.userInfo, session.confirmedEmail, userInfo.eligible, session.bankDetails))
```

Comment:
why are we validating `userInfo.userInfo.email` when we're not even using it here? I think I remember that the email in `EligibleWithEmail` was intended to be the email in userinfo if it existed not the confirmed email in the session

Reply:
opps i missed it, yeah we are not using the `validEmail` here, so we don't need to validate it , also by this time the user should have a `confirmedEmail` in the session